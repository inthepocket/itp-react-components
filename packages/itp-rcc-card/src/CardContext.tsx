import { createContext } from 'react';
import CardStylesInterface from './CardStylesInterface';

interface CardContextInterface {
  prefixCss?: string | null;
  size: 'small' | 'default' | 'large';
  styles?: CardStylesInterface | null;
}

/**
 * Card Context
 */
const CardContext =
  createContext<CardContextInterface>({ prefixCss: 'card', size: 'default', styles: null });

export default CardContext;
