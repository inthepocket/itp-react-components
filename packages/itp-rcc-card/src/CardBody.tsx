import * as React from 'react';
import capitalize from 'capitalize';
import classnames from 'classnames';
import CardContext from './CardContext';
import CardChildPropsInterface from './CardChildPropsInterface';

const { Consumer } = CardContext;

/**
 * Card Body
 */
const CardBody = ({ children }: CardChildPropsInterface) => {
  return (
    <Consumer>
      {(cardContext) => {
        const { prefixCss, size, styles } = cardContext;
        const containerClassName = styles ? styles.body : `${prefixCss}-body`;
        const sizeClassName = styles ? styles[`bodySize${capitalize(size)}`] : `${prefixCss}-body-size-${size}`;

        const className = classnames({
          [`${containerClassName}`]: containerClassName,
          [`${sizeClassName}`]: sizeClassName,
        });

        return (
          <div className={className}>{children}</div>
        );
      }}
    </Consumer>
  );
};

export default CardBody;
