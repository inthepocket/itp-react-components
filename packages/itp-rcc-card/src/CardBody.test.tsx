import * as React from 'react';
import { create } from 'react-test-renderer';
import CardBody from './CardBody';
import CardContext from './CardContext';

const { Provider } = CardContext;
const cssModuleMock = {
  body: 'cssModulesBody',
  bodySizeSmall: 'cssModulesBodySizeSmall',
  bodySizeLarge: 'cssModulesBodySizeLarge',
  card: 'cssModulesCard',
};

test('CardBody: default', () => {
  const tree = create((
    <CardBody>
      <p>Cras sed ante. Phasellus in massa. Curabitur dolor eros, gravida et, hendrerit ac, cursus non, massa. Aliquam lorem.</p>
    </CardBody>
  ))
  .toJSON();

  expect(tree).toMatchSnapshot();
});

test('CardBody: size: small', () => {
  const tree = create((
    <Provider value={{ prefixCss: 'card', size: 'small' }}>
      <CardBody>
        <p>Cras sed ante. Phasellus in massa. Curabitur dolor eros, gravida et, hendrerit ac, cursus non, massa. Aliquam lorem.</p>
      </CardBody>
    </Provider>
  ))
  .toJSON();

  expect(tree).toMatchSnapshot();
});

test('CardBody: size: large', () => {
  const tree = create((
    <Provider value={{ prefixCss: 'card', size: 'large' }}>
      <CardBody>
        <p>Cras sed ante. Phasellus in massa. Curabitur dolor eros, gravida et, hendrerit ac, cursus non, massa. Aliquam lorem.</p>
      </CardBody>
    </Provider>
  ))
  .toJSON();

  expect(tree).toMatchSnapshot();
});

test('CardBody: css modules: default', () => {
  const tree = create((
    <Provider value={{ prefixCss: 'card', size: 'default', styles: cssModuleMock }}>
      <CardBody>
        <p>Cras sed ante. Phasellus in massa. Curabitur dolor eros, gravida et, hendrerit ac, cursus non, massa. Aliquam lorem.</p>
      </CardBody>
    </Provider>
  ))
  .toJSON();

  expect(tree).toMatchSnapshot();
});

test('CardBody: css modules: size: small', () => {
  const tree = create((
    <Provider value={{ prefixCss: 'card', size: 'small', styles: cssModuleMock }}>
      <CardBody>
        <p>Cras sed ante. Phasellus in massa. Curabitur dolor eros, gravida et, hendrerit ac, cursus non, massa. Aliquam lorem.</p>
      </CardBody>
    </Provider>
  ))
  .toJSON();

  expect(tree).toMatchSnapshot();
});

test('CardBody: css modules: size: large', () => {
  const tree = create((
    <Provider value={{ prefixCss: 'card', size: 'large', styles: cssModuleMock }}>
      <CardBody>
        <p>Cras sed ante. Phasellus in massa. Curabitur dolor eros, gravida et, hendrerit ac, cursus non, massa. Aliquam lorem.</p>
      </CardBody>
    </Provider>
  ))
  .toJSON();

  expect(tree).toMatchSnapshot();
});