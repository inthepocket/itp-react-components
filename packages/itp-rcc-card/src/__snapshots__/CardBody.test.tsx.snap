// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardBody: css modules: default 1`] = `
<div
  className="cssModulesBody"
>
  <p>
    Cras sed ante. Phasellus in massa. Curabitur dolor eros, gravida et, hendrerit ac, cursus non, massa. Aliquam lorem.
  </p>
</div>
`;

exports[`CardBody: css modules: size: large 1`] = `
<div
  className="cssModulesBody cssModulesBodySizeLarge"
>
  <p>
    Cras sed ante. Phasellus in massa. Curabitur dolor eros, gravida et, hendrerit ac, cursus non, massa. Aliquam lorem.
  </p>
</div>
`;

exports[`CardBody: css modules: size: small 1`] = `
<div
  className="cssModulesBody cssModulesBodySizeSmall"
>
  <p>
    Cras sed ante. Phasellus in massa. Curabitur dolor eros, gravida et, hendrerit ac, cursus non, massa. Aliquam lorem.
  </p>
</div>
`;

exports[`CardBody: default 1`] = `
<div
  className="card-body card-body-size-default"
>
  <p>
    Cras sed ante. Phasellus in massa. Curabitur dolor eros, gravida et, hendrerit ac, cursus non, massa. Aliquam lorem.
  </p>
</div>
`;

exports[`CardBody: size: large 1`] = `
<div
  className="card-body card-body-size-large"
>
  <p>
    Cras sed ante. Phasellus in massa. Curabitur dolor eros, gravida et, hendrerit ac, cursus non, massa. Aliquam lorem.
  </p>
</div>
`;

exports[`CardBody: size: small 1`] = `
<div
  className="card-body card-body-size-small"
>
  <p>
    Cras sed ante. Phasellus in massa. Curabitur dolor eros, gravida et, hendrerit ac, cursus non, massa. Aliquam lorem.
  </p>
</div>
`;
